docs/proposals/secret-configmap-file-mode.md

```diff
@@ -0,0 +1,111 @@
+it needs the ssh keys, and ssh will only work if the ssh key file mode is the
+one it expects. 
+
+This was asked on the mailing list here[2] and here[3], too.
+
+[1]: https://www.postgresql.org/docs/9.1/static/libpq-pgpass.html
+[2]: https://groups.google.com/forum/#!topic/kubernetes-dev/eTnfMJSqmaM
+[3]: https://groups.google.com/forum/#!topic/google-containers/EcaOPq4M758
+
+# Alternatives considered
+
+Several alternatives have been considered:
+
+ * Add a mode to the API definition when using secrets: this is backward
+   compatible as described in (docs/devel/api_changes.md) IIUC and seems like the
+   way to go. Also @thockin said in the ML that he would consider such an
+   approach. But it might be worth to consider if we want to do the same for
+   configmaps or owners, but there is no need to do it now either.
+
+ * Change the default file mode for secrets: I think this is unacceptable as it
+   is stated in the api_changes doc. And besides it doesn't feel correct IMHO, it
+   is technically one option. The argument for this might be that world and group
+   readable for a secret is not a nice default, we already take care of not
+   writing it to disk, etc. but the file is created world-readable anyways. Such a
+   default change has been done recently: the default was 0444 in kubernetes <= 1.2
+   and is now 0644 in kubernetes >= 1.3 (and the file is not a regular file,
+   it's a symlink now). This change was done here to minimize differences between
+   configmaps and secrets: https://github.com/kubernetes/kubernetes/pull/25285. But
+   doing it again, and changing to something more restrictive (now is 0644 and it
+   should be 0400 to work with ssh and most apps) seems too risky, it's even more
+   restrictive than in k8s 1.2. Specially if there is no way to revert to the old
+   permissions and some use case is broken by this. And if we are adding a way to
+   change it, like in the option above, there is no need to rush changing the
+   default. So I would discard this.
+
+ * We don't want to people be able to change this, at least for now, and the
+   ones who do, suggest that do it as a "postStart" command. This is acceptable
+   if we don't want to change kubernetes core for some reason, although there
+   seem to be valid use cases. But if the user want's to use the "postStart" for
+   something else, then it is more disturbing to do both things (have a script
+   in the docker image that deals with this, but is not probably concern of the
+   project so it's not nice, or specify several commands by using "sh").
+
+# Proposed implementation
+
+The proposed implementation goes with the first alternative: adding a `mode` to
+the API.
+
+The proposition is to add it as a field in the KeyToPath. That is, as
+v1.SecretVolumeSource in the API[5] and the equivalent for configmaps[4]. And
+then change the key to be optional, but is mandatory to specify at least one of
```

Comment:
I was thinking about the Path, sorry. But yes, you are right, it may break assumptions doing that.

What I had in mind was:
If you create a secret/configmap from a dir, for example, you may just want to change the permissions on a key file. But having to specify the path, if it is the same as the key and you don't need anything but the identity mapping, is disturbing.

Your suggestion seems better. Although, I'm not 100% happy with having to specify the same path as the key, if you just want to change the permissions of one file.

Do you have any other great idea? :)